spec/index.bs

```diff
@@ -2521,8 +2532,10 @@ enum GPUBindingType {
     "storage-buffer",
     "readonly-storage-buffer",
     "sampler",
+    "filtering-sampler",
```

Comment:
What about calling `sampler` being filtering and `addressing-sampler` being non-filtering? This way the simpler wording is for the most common thing to use.

Ditto for `sampled-texture` being filtering and `sampler-addressed-texture` being non-filtered? (I'm less happy about this one).

Reply:
I'm not attached to the current names. I do find "addressing" part confusing though. We have sampler addressing modes, so seeing "addressing-sampler" seems weird.

There is one consideration to make when looking for a default/non-default distinction:
  - "filtering-sampler" is more *limited* versus "sampler", since it only works with a subset of formats
  - "sampled-texture"  is more *limited* versus "filtered-texture", since it only works with a subset of samplers...

So currently, the "filtering-" means restrictions in one place, and "filtered-" means more capability in another, which is a bit strange. It may be not too bad when we don't have a clear default, i.e. "sampled-texture" vs "filtered-texture".

Reply:
> "filtering-sampler" is more _limited_ versus "sampler"

I don't think of it this way. It fits with a more limited set of texture types in the shader, but the set of values you can put in the slot is less limited, so it's a supertype. (The compatible texture type is contravariant to the sampler type.) So I think I would prefer that the subtype have more adjectives:

- BGL "sampler" can be a non-filtering sampler or filtering sampler
- BGL "nonfiltering-sampler" ("addressing-sampler"?) can be a non-filtering sampler only

> "sampled-texture"  is more _limited_ versus "filtered-texture"

"sampled texture" is a supertype of "filtered texture". Same logic as above:

- BGL "sampled-texture" can be a filterable or non-filterable texture
- BGL "filtered-sampled-texture" ("filtered-texture"?) can be only filterable textures

Reply:
My concern about this is that your 2 defaults are not compatible, i.e. using `"sampled-texture" with "sampler" is not allowed, because it's not guaranteed that the texture will be filterable. So what's the point of introducing defaults if they lead you into a corner, and end up producing an error anyway?

Reply:
That's a fair point, but I also don't like the wording of "filtering-sampler", which sounds like filtering-only, but is actually either filtering or non-filtering.

Reply:
How about "sampler"/"point-sampler" and "filtered-texture"/"sampled-texture"? This is basically reversing the naming for samplers versus what the PR has now.

Reply:
I don't think it's particularly more clear than "nonfiltering-sampler", though it's fine. It still doesn't sound like the "default" to me though; why does it satisfy the concern you had with "nonfiltering-sampler"?

Reply:
In the version you suggested there were 2 defaults, and they didn't work with each other. In the version here there is only a "default" for the sampler, but texture bindings are even. So that makes the concern less of a problem.

Reply:
That possibility was included in my suggestion, and it would be fine with me. "nonfiltering-sampler" still seems clearer to me.

Reply:
What about `nearest-sampler`? It is very clear in relation to the sampler creation parameters, and we could have a `nearest-sampler-texture`?

Reply:
My concern with `nearest-sampled-texture` approach is that the default (which is implied to be `sampled-texture`) wouldn't work for half of the texture formats. Why would we want it to be shorter to type or preferred then?

Reply:
My intuition is that filtered textures are the most common by far, and that integer of float32 textures are maybe 10-20% of all bindings usage. So it would make sense for filterable sampled texture to be the "default" and be `sampled-texture`. However I don't have data to back that up :/

Reply:
I think it's more important to be clear to code readers than writers. A developer writing this code will get a clear error message if they use a texture or sampler with layout type that doesn't match it.

Reply:
ok, I find your words convincing :)